ViewModels/PackCreationViewModel.cs

```diff
@@ -0,0 +1,345 @@
+            set
+            {
+                _description = value;
+                OnPropertyChanged(nameof(Description));
+            }
+        }
+
+        private readonly int CardLimit = 10;
+        private Dictionary<int, int> _rarityProbabilities;
+
+
+
+        public PackCreationViewModel(string packType)
+        {
+            _backendService = new BackendService(ClientSingleton.GetInstance("http://api.combatcritters.ca:4000"));
+            _packType = packType;
+            _description = "";
+            _gameCards = new ObservableCollection<ICard>();
+            _selectedCards = new ObservableCollection<ICard>();
+            HasCards = false;
+
+            SelectionChangedCommand = new Command<ICard>(OnCardSelected);
+            CreatePackCommand = new Command(async () => await CreatePackAsync());
+
+
+
+            // Initialize raritylimits and rarity probabilities based on initial PackType
+            SetRarityLimits(PackType);
+            _rarityProbabilities = new Dictionary<int, int>
+            {
+                { (int)Rarity.COMMON, 0 },
+                { (int)Rarity.UNCOMMON, 0 },
+                { (int)Rarity.RARE, 0 },
+                { (int)Rarity.EPIC, 0 },
+                { (int)Rarity.LEGENDARY, 0 }
+            };
+            SetRarityProbabilities(PackType);
+
+            //start Loading the user cards.
+            Task.Run(async () => await InitializeViewModelAsync());
+
+        }
+
+        private async Task InitializeViewModelAsync()
+        {
+            await LoadGameCards();
+        }
+
+        /// <summary>
+        /// Load all available game cards using backend services
+        /// </summary>
+        /// <returns>Task</returns>
+        public async Task LoadGameCards()
+        {
+            IsLoading = true;
+            bool hasCards = false; // function scoped variable
+
+            try
+            {
+                CardQueryBuilder filteredBuild = new();
+                var cards = await _backendService.GetCardsAsync(filteredBuild.Build());
+
+                if (cards != null && cards.Count > 0)
+                {
+                    GameCards = new ObservableCollection<ICard>(cards.Select(stack => stack.Item).ToList());
+                    hasCards = true;
+                    Console.WriteLine($"Number of cards loaded: {GameCards.Count}");
+                }
+                else
+                {
+                    //Game has no Cards
+                    GameCards.Clear();
+                }
+            }
+            catch (RestException)
+            {
+                if (Application.Current?.MainPage != null)
+                    await Application.Current.MainPage.DisplayAlert("Error", "Failed to load game cards. Please try again.", "OK");
+            }
+
+            finally
+            {
+                //Set HasCards based on result of operation
+                HasCards = hasCards;
+                IsLoading = false;
+            }
+        }
```

Comment:
this is duplicate code, I believe its in the cards view. not totally sure though